rpc/gnmi/protobuf-vals.md

```diff
@@ -0,0 +1,214 @@
+# Carrying Binary Wire Format Protobuf Messages in gNMI
+**Contributors**: robjs, csl  
+May 2018  
+*Implementation Status*: Proposal
+
+## Problem Statement
+For some applications the data carried by gNMI is directly described by a
+Protobuf IDL file which has been auto-generated from a YANG schema. Elements
+within the Protobuf IDL are therefore 1:1 mapped to those within the YANG
+schema - with the mapping between the YANG tree node and the protobuf
+entity being established by the schema path of each.
+
+When carrying such protobufs on the wire, it is undesirable to encode the
+message name and namespace into a `protobuf.Any` field, as this:
+
+ * Causes duplication of data on the wire - since the gNMI Notification's path
+   (the concatenated `path` and `prefix` fields) already indicates which message
+   is serialised within the Notification.
+
+ * Results in compatibilities when generated protobufs are within different
+   namespaces, but represent the same YANG schema. This is typically the case
+   when generating an augmented version of a standard schema. In this case, the
+   augmented messages are backwards compatible with the standard messages, since
+   they are a strict superset of the fields, and hence can be deserialised into
+   the "standard" message. The differing type in protobuf.Any does not allow
```

Comment:
add backticks to protobuf.Any as above ?

Reply:
Done.